src/pages/search/Search.tsx

```diff
@@ -29,11 +30,38 @@ const Search = () => {
   );
   return (<Fade>
     <Query query={SEARCH_LISTINGS} variables={{ input, ...LISTING_CARD_IMAGE_DIMENSIONS }}>
-      {({ loading, error, data }) =>
-        loading ? <Loading /> :
-        error ? <h2>Error: {error.message}</h2> :
-        <SearchPage listings={data.searchListings} />
-      }
+      {({ loading, error, data }) => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
+          );
+        }
+        if (error && error.message.includes('400') && Object.keys(queryParams).length === 0) {
```

Comment:
this is bad: `error.message.includes('400')`

Reply:
apollo errors has an `invalidArgs` parameter you can use.

Reply:
![screen shot 2019-03-05 at 1 10 42 pm](https://user-images.githubusercontent.com/7976/53837719-1a1fd600-3f48-11e9-88a3-44385a7c7d1f.png)

i'm not sure this works either. I did an empty search and got an error instead of the search bar.

Reply:
ah it's because there's a 1 for guests, ill add a check for that